server/v2/core/store/types.go

```diff
@@ -13,20 +13,48 @@ type Store interface {
 	// StateLatest returns a readonly view over the latest
 	// committed state of the store. Alongside the version
 	// associated with it.
-	StateLatest() (uint64, ReadonlyState, error)
+	StateLatest() (uint64, ReadonlyAccountsState, error)
 
 	// StateAt returns a readonly view over the provided
 	// state. Must error when the version does not exist.
-	StateAt(version uint64) (ReadonlyState, error)
+	StateAt(version uint64) (ReadonlyAccountsState, error)
 
 	// StateCommit commits the provided changeset and returns
 	// the new state root of the state.
-	StateCommit(changes []ChangeSet) (Hash, error)
+	StateCommit(changes []AccountStateChanges) (Hash, error)
 }
 
-// ChangeSet represents a change in a key and value of state.
+// ReadonlyAccountsState represents a readonly view over all the accounts state.
+type ReadonlyAccountsState interface {
+	// GetAccountReadonlyState must return the state for the provided address.
+	// Storage implements might treat this as a prefix store over an address.
+	// Prefix safety is on the implementer.
+	GetAccountReadonlyState(address []byte) (ReadonlyState, error)
+}
+
+// WritableAccountsState represents a writable account state.
+type WritableAccountsState interface {
+	ReadonlyAccountsState
+	// GetAccountWritableState must the return a WritableState
+	// for the provided account address.
+	GetAccountWritableState(address []byte) (WritableState, error)
+	// ApplyAccountsStateChanges applies all the state changes
+	// of the accounts. Ordering of the returned state changes
+	// is an implementation detail and must not be assumed.
+	ApplyAccountsStateChanges(stateChanges []AccountStateChanges) error
```

Comment:
```suggestion
	ApplyStateChanges(stateChanges []AccountStateChanges) error
```

dont think we need accounts everywhere